src/rgw/rgw_common.h

```diff
@@ -264,6 +264,45 @@ enum RGWObjCategory {
   RGW_OBJ_CATEGORY_MULTIMETA = 3,
 };
 
+
+class rgw_ret {
+  int ret;
+  boost::optional<std::string> message;
+
+  rgw_ret(const int ret, std::string message)
+    : ret(ret),
+      message(std::move(message)) {
+  }
+
+public:
+  rgw_ret()
+    : ret(0) {
+  }
+
+  rgw_ret(const int ret)
+    : ret(ret) {
+  }
+
+  rgw_ret& operator=(const int ret) {
+    this->ret = ret;
+    return *this;
+  }
+
+  operator int() const {
+    return ret;
+  }
+
+  boost::optional<std::string> get_err_msg() const {
+    return message;
+  }
+
+  /* Make a ret value with a customized status message. */
+  static rgw_ret make_custom(const int ret, std::string message) {
+    return rgw_ret(ret, std::move(message));
+  }
+};
+
```

Comment:
We should not use this class, we should use std::system_error. Yes, normally it's thrown as an exception (and we might want to do that at some point) but it provides exactly what we want here. It combines an error code and an optional string.

The std::error_code is also a good candidate for replacing integers since it allows us to keep SSL errors, SWIFT errors, S3 errors, RADOS errors and all the rest in separate families, but have a single test against 'Object not found' that works for all of them.